src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -557,6 +557,15 @@ DistributedInsertSelectSupported(Query *queryTree, RangeTblEntry *insertRte,
 							 NULL, NULL);
 	}
 
+	List *subqueryRteList = ExtractRangeTableEntryList(subquery);
+	if (CheckRTEList(subqueryRteList, IsCitusLocalTableRTE))
```

Comment:
Why don't we use  `FindNodeMatchingCheckFunction(subquery, IsCitusLocalTableRTE)`?

Also, why don't we do the same  for Postgres local tables? I think we should at least have a comment explaining why even if we have a good reason

Reply:
just added it for completeness in https://github.com/citusdata/citus/pull/4133/commits/9016ec2cf6d55de670724dab06e45e770ca7b48e but it didn't change any test outputs ?

Can you think of a test case that would error out in that check ?